### PR TITLE
Tara testing

### PR DESCRIPTION
**Pull request by tmh528**

### CODE REVIEW

File: core/src/test/java/PlayerTests.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    @Test
+    public void FreezeUpTest() {
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get(2));
+        player = new Player(sprite, playerCollisionLayers, true);
+        mapGraph.addNode(node);
+        mapGraph.addNode(node2);
+        mapGraph.connectNodes(node, node2);
+
+        frozen_infiltrator = new Infiltrator(sprite, node, mapGraph);
```
Comment (dallinson) [inline core/src/test/java/PlayerTests.java]: This is causing a null pointer exception

Comment (tmh528) [reply core/src/test/java/PlayerTests.java]: > fixed!

File: core/src/test/java/DifficultyTests.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@RunWith(GdxTestRunner.class)
+public class DifficultyTests {
+    @Before
+    public void before(){
+        Gdx.gl = Mockito.mock(GL20.class);
+        //SpriteBatch = Mockito.mock
+    }
+
+    @Test
+    public void IncorrectArrestsTest() {
```
Comment (aimiebell) [inline core/src/test/java/DifficultyTests.java:29]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/test/java/InfiltratorTests.java [hunk 3]
```diff
@@ -53,11 +52,8 @@ public void NumberOfInfiltratorsTest(){
     public void InfiltratorDamagePlayer(){
```
Comment (aimiebell) [inline core/src/test/java/InfiltratorTests.java:52]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/test/java/PlayerTests.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    GraphCreator graphCreator = new GraphCreator( (TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+
+    Infiltrator frozen_infiltrator, infiltrator_damage_system;
+    CrewMembers not_frozen_crew;
+
+    /**
+     * Tests that players can take damage
+     */
+    @Test
+    public void TakeDamageTest() {
```
Comment (aimiebell) [inline core/src/test/java/PlayerTests.java:58]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/test/java/DifficultyTests.java (hunk 2 above)
Comment (aimiebell) [inline core/src/test/java/DifficultyTests.java:29]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/test/java/InfiltratorTests.java (hunk 3 above)
Comment (aimiebell) [inline core/src/test/java/InfiltratorTests.java:52]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/test/java/PlayerTests.java (hunk 4 above)
Comment (aimiebell) [inline core/src/test/java/PlayerTests.java:58]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request merged**